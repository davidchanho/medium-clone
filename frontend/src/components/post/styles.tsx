import styled from "styled-components";

export const Container = styled.div``;

export const Header = styled.header`
  display: flex;
  align-items: center;
`;

export const Icon = styled.img`
  border-radius: 50%;
`;
