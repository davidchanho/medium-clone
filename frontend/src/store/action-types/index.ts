export enum ActionTypes {
  FETCH_POSTS = "FETCH_POSTS",
  FETCH_POSTS_SUCCESS = "FETCH_POSTS_SUCCESS",
  FETCH_POSTS_FAIL = "FETCH_POSTS_FAIL",
  FETCH_POST = "FETCH_POST",
  FETCH_POST_SUCCESS = "FETCH_POST_SUCCESS",
  FETCH_POST_FAIL = "FETCH_POST_FAIL",
  ADD_POST = "ADD_POST",
  DELETE_POST = "DELETE_POST",
  UPDATE_POST = "UPDATE_POST",
}
