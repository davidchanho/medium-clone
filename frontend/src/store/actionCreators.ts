import { Dispatch } from "react";
import { Action, ActionTypes } from ".";
import { IComment, IPost, IUser } from "../types";
import API from "./api";

export const getComments = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_COMMENTS });

  try {
    const data = await API.getComments();
    dispatch({
      type: ActionTypes.FETCH_COMMENTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_COMMENTS_FAIL,
      payload: err,
    });
  }
};

export const getComment = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_COMMENT });

  try {
    const data = await API.getComment(_id);
    dispatch({
      type: ActionTypes.FETCH_COMMENT_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_COMMENT_FAIL,
      payload: err,
    });
  }
};

export const addComment = (comment: IComment) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.addComment(comment);
    dispatch({ type: ActionTypes.ADD_COMMENT, payload: comment });
  } catch (err) {
    console.log(err);
  }
};

export const deleteComment = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.deleteComment(_id);
    dispatch({
      type: ActionTypes.DELETE_COMMENT,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updateComment = (comment: IComment) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updateComment(comment);
    dispatch({
      type: ActionTypes.UPDATE_COMMENT,
      payload: comment,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getPosts = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POSTS });
  try {
    const data = await API.getPosts();

    dispatch({
      type: ActionTypes.FETCH_POSTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POSTS_FAIL,
      payload: err,
    });
  }
};

export const getPost = (_id: string) => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POST });

  try {
    const data = await API.getPost(_id);

    dispatch({
      type: ActionTypes.FETCH_POST_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POST_FAIL,
      payload: err,
    });
  }
};

export const addPost = (post: IPost) => async (dispatch: Dispatch<Action>) => {
  try {
    API.addPost(post);

    dispatch({ type: ActionTypes.ADD_POST, payload: post });
  } catch (err) {
    console.log(err);
  }
};

export const dismissPost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    dispatch({
      type: ActionTypes.DISMISS_POST,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updatePost(post);

    dispatch({
      type: ActionTypes.UPDATE_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getPublications = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATIONS });

  try {
    const data = await API.getPublications();
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_FAIL,
      payload: err,
    });
  }
};

export const getPublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATION });

  try {
    const data = await API.getPublication(_id);
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_FAIL,
      payload: err,
    });
  }
};

export const getUsers = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_USERS });

  try {
    const data = await API.getUsers();
    dispatch({
      type: ActionTypes.FETCH_USERS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_USERS_FAIL,
      payload: err,
    });
  }
};

export const getUser = (_id: string) => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_USER });

  try {
    const data = await API.getUser(_id);
    dispatch({
      type: ActionTypes.FETCH_USER_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_USER_FAIL,
      payload: err,
    });
  }
};

export const addUser = (user: IUser) => async (dispatch: Dispatch<Action>) => {
  try {
    API.addUser(user);
    dispatch({ type: ActionTypes.ADD_USER, payload: user });
  } catch (err) {
    console.log(err);
  }
};

export const deleteUser = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.deleteUser(_id);
    dispatch({
      type: ActionTypes.DELETE_USER,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updateUser = (user: IUser) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    API.updateUser(user);
    dispatch({
      type: ActionTypes.UPDATE_USER,
      payload: user,
    });
  } catch (err) {
    console.log(err);
  }
};

export const bookmarkPost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    dispatch({
      type: ActionTypes.BOOKMARK_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const getTopics = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_TOPICS });

  try {
    const data = await API.getTopics();
    dispatch({
      type: ActionTypes.FETCH_TOPICS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_TOPICS_FAIL,
      payload: err,
    });
  }
};
