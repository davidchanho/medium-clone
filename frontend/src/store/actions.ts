import { ActionTypes } from ".";
import { IComment, IItem, IPost, IPublication, ITopic, IUser } from "../types";

export interface LoadingItemsAction {
  type: ActionTypes.LOADING_ITEMS;
}

export interface ErrorItemsAction {
  type: ActionTypes.ERROR_ITEMS;
  name: string;
  payload: string;
}

export interface FetchItemsAction {
  type: ActionTypes.FETCH_ITEMS;
  name: string;
  payload: IItem[];
}

export interface FetchItemAction {
  type: ActionTypes.FETCH_ITEM;
  name: string;
  payload: IItem;
}

export interface AddItemAction {
  type: ActionTypes.ADD_ITEM;
  name: string;
  payload: IItem;
}

export interface DeleteItemAction {
  type: ActionTypes.DELETE_ITEM;
  name: string;
  payload: string;
}

export interface UpdateItemAction {
  type: ActionTypes.UPDATE_ITEM;
  name: string;
  payload: IItem;
}

export interface LoadingCommentsAction {
  type: ActionTypes.LOADING_COMMENTS;
}

export interface ErrorCommentsAction {
  type: ActionTypes.ERROR_COMMENTS;
  payload: string;
}

export interface FetchCommentsAction {
  type: ActionTypes.FETCH_COMMENTS;
  payload: IComment[];
}

export interface FetchCommentAction {
  type: ActionTypes.FETCH_COMMENT;
  payload: IComment;
}

export interface AddCommentAction {
  type: ActionTypes.ADD_COMMENT;
  payload: IComment;
}

export interface DeleteCommentAction {
  type: ActionTypes.DELETE_COMMENT;
  payload: string;
}

export interface UpdateCommentAction {
  type: ActionTypes.UPDATE_COMMENT;
  payload: IComment;
}

export interface LoadingPostsAction {
  type: ActionTypes.LOADING_POSTS;
}

export interface FetchPostsAction {
  type: ActionTypes.FETCH_POSTS;
  payload: IPost[];
}

export interface ErrorPostsAction {
  type: ActionTypes.ERROR_POSTS;
  payload: string;
}

export interface FetchPostAction {
  type: ActionTypes.FETCH_POST;
  payload: IPost;
}

export interface AddPostAction {
  type: ActionTypes.ADD_POST;
  payload: IPost;
}

export interface DeletePostAction {
  type: ActionTypes.DELETE_POST;
  payload: string;
}

export interface UpdatePostAction {
  type: ActionTypes.UPDATE_POST;
  payload: IPost;
}

export interface LoadingPublicationsAction {
  type: ActionTypes.LOADING_PUBLICATIONS;
}

export interface FetchPublicationsAction {
  type: ActionTypes.FETCH_PUBLICATIONS;
  payload: IPublication[];
}

export interface ErrorPublicationsAction {
  type: ActionTypes.ERROR_PUBLICATIONS;
  payload: string;
}

export interface FetchPublicationAction {
  type: ActionTypes.FETCH_PUBLICATION;
  payload: IPublication;
}

export interface LoadingTopicsAction {
  type: ActionTypes.LOADING_TOPICS;
}

export interface FetchTopicsAction {
  type: ActionTypes.FETCH_TOPICS;
  payload: ITopic[];
}

export interface ErrorTopicsAction {
  type: ActionTypes.ERROR_TOPICS;
  payload: string;
}

export interface LoadingUsersAction {
  type: ActionTypes.LOADING_USERS;
}

export interface FetchUsersAction {
  type: ActionTypes.FETCH_USERS;
  payload: IUser[];
}

export interface ErrorUsersAction {
  type: ActionTypes.ERROR_USERS;
  payload: string;
}

export interface FetchUserAction {
  type: ActionTypes.FETCH_USER;
  payload: IUser;
}

export interface AddUserAction {
  type: ActionTypes.ADD_USER;
  payload: IUser;
}

export interface DeleteUserAction {
  type: ActionTypes.DELETE_USER;
  payload: string;
}

export interface UpdateUserAction {
  type: ActionTypes.UPDATE_USER;
  payload: IUser;
}

export interface BookmarkPostAction {
  type: ActionTypes.BOOKMARK_POST;
  payload: IPost;
}

export type Action =
  | FetchItemsAction
  | ErrorItemsAction
  | LoadingItemsAction
  | FetchItemAction
  | AddItemAction
  | DeleteItemAction
  | UpdateItemAction
  | FetchCommentsAction
  | ErrorCommentsAction
  | LoadingCommentsAction
  | FetchCommentAction
  | AddCommentAction
  | DeleteCommentAction
  | UpdateCommentAction
  | FetchPostsAction
  | ErrorPostsAction
  | LoadingPostsAction
  | FetchPostAction
  | AddPostAction
  | DeletePostAction
  | UpdatePostAction
  | FetchPublicationsAction
  | ErrorPublicationsAction
  | LoadingPublicationsAction
  | FetchPublicationAction
  | FetchTopicsAction
  | ErrorTopicsAction
  | LoadingTopicsAction
  | FetchUsersAction
  | ErrorUsersAction
  | LoadingUsersAction
  | FetchUserAction
  | AddUserAction
  | DeleteUserAction
  | UpdateUserAction
  | BookmarkPostAction;
