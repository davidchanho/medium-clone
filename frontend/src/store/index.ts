export * as actionCreators from "./actionCreators";
export * from "./actions";
export * from "./actionTypes";
export * from "./api";
export * from "./reducers";
export * from "./selectors";
export * from "./store";
