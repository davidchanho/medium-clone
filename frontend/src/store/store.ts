import localForage from "localforage";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createLogger } from "redux-logger";
import { persistReducer, persistStore } from "redux-persist";
import thunk from "redux-thunk";
import reducers from "./reducers";

// localForage
localForage.config({
  driver: localForage.INDEXEDDB,
});

// redux-perist
const persistConfig = {
  key: "medium",
  storage: localForage,
  whitelist: ["posts"],
};
const persistedReducer = persistReducer(persistConfig, reducers);

// middleware
const logger = createLogger();
const middleware = [thunk, logger];

export const store = createStore(
  persistedReducer,
  {},
  composeWithDevTools(applyMiddleware(...middleware))
);

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
