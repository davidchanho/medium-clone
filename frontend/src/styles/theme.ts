import * as breakpoints from "./breakpoints";
import * as colors from "./colors";

export const theme = () => ({
  breakpoints: {
    mobile: breakpoints.mobile,
    tablet: breakpoints.tablet,
    desktop: breakpoints.desktop,
  },
  colors: {
    primaryButton: colors.white["100"],
    primaryButtonBorder: colors.grey["100"],
    primaryButtonText: colors.grey["100"],

    secondaryText: colors.grey["100"],
    secondaryTextHover: colors.black["200"],

    numberedText: colors.grey["200"],

    successButton: colors.green["100"],
    successButtonHover: colors.green["200"],
    successButtonText: colors.white["100"],

    upgradePrimary: colors.grey["200"],
    upgradePrimaryHover: colors.green["100"],

    successText: colors.green["100"],

    darkBackground: colors.black["200"],
    darkText: colors.black["200"],

    white: colors.white["100"],
  },
});
