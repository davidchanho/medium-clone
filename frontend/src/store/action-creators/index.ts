import { Dispatch } from "redux";
import db from "../../api";
import { ActionTypes } from "../action-types";
import { Action } from "../actions";
import { IPost, IPublication } from "../reducers/posts";

export const fetchPosts = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_POSTS });

  try {
    const data = await db.fetchPosts();
    dispatch({
      type: ActionTypes.FETCH_POSTS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POSTS_FAIL,
      payload: err,
    });
  }
};

export const fetchPost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_POST });

  try {
    const data = await db.fetchPost(_id);
    dispatch({
      type: ActionTypes.FETCH_POST_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_POST_FAIL,
      payload: err,
    });
  }
};

export const addPost = (post: IPost) => async (dispatch: Dispatch<Action>) => {
  try {
    db.addPost(post);
    dispatch({ type: ActionTypes.ADD_POST, payload: post });
  } catch (err) {
    console.log(err);
  }
};

export const deletePost = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.deletePost(_id);
    dispatch({
      type: ActionTypes.DELETE_POST,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePost = (post: IPost) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.updatePost(post);
    dispatch({
      type: ActionTypes.UPDATE_POST,
      payload: post,
    });
  } catch (err) {
    console.log(err);
  }
};

export const fetchPublications = () => async (dispatch: Dispatch<Action>) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATIONS });

  try {
    const data = await db.fetchPublications();
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATIONS_FAIL,
      payload: err,
    });
  }
};

export const fetchPublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  dispatch({ type: ActionTypes.FETCH_PUBLICATION });

  try {
    const data = await db.fetchPublication(_id);
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_SUCCESS,
      payload: data,
    });
  } catch (err) {
    dispatch({
      type: ActionTypes.FETCH_PUBLICATION_FAIL,
      payload: err,
    });
  }
};

export const addPublication = (publication: IPublication) => async (dispatch: Dispatch<Action>) => {
  try {
    db.addPublication(publication);
    dispatch({ type: ActionTypes.ADD_PUBLICATION, payload: publication });
  } catch (err) {
    console.log(err);
  }
};

export const deletePublication = (_id: string) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.deletePublication(_id);
    dispatch({
      type: ActionTypes.DELETE_PUBLICATION,
      payload: _id,
    });
  } catch (err) {
    console.log(err);
  }
};

export const updatePublication = (publication: IPublication) => async (
  dispatch: Dispatch<Action>
) => {
  try {
    db.updatePublication(publication);
    dispatch({
      type: ActionTypes.UPDATE_PUBLICATION,
      payload: publication,
    });
  } catch (err) {
    console.log(err);
  }
};
