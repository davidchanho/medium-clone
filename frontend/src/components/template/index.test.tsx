import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import Template from ".";
import { store } from "../../store";
import { theme } from "../../styles/theme";

describe("Template", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeProvider theme={theme}>
            <Router>
              <Template />
            </Router>
          </ThemeProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
