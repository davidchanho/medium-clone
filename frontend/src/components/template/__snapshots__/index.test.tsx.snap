// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template renders correctly 1`] = `
Array [
  <div>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        height="34"
        width="180"
      >
        medium.svg
      </svg>
    </a>
     
    Good evening
    <div
      aria-controls="basic-navbar-nav"
    />
    <div
      id="basic-navbar-nav"
    >
      <nav
        className="sc-furvIG hwJvNN"
      >
        <ul>
          <li>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                fillRule="evenodd"
              />
              <path
                clipRule="evenodd"
                d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                fillRule="evenodd"
              />
            </svg>
          </li>
          <li>
            <a
              className=""
              href="/reading-list"
              onClick={[Function]}
              style={Object {}}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M7 13l5 3V4a2 2 0 00-2-2H4a2 2 0 00-2 2v12l5-3zm-4 1.234l4-2.4 4 2.4V4a1 1 0 00-1-1H4a1 1 0 00-1 1v10.234z"
                  fillRule="evenodd"
                />
                <path
                  d="M14 14l-1-.6V2a1 1 0 00-1-1H4.268A2 2 0 016 0h6a2 2 0 012 2v12z"
                />
              </svg>
            </a>
          </li>
          <li>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 16a2 2 0 002-2H6a2 2 0 002 2z"
              />
              <path
                clipRule="evenodd"
                d="M8 1.918l-.797.161A4.002 4.002 0 004 6c0 .628-.134 2.197-.459 3.742-.16.767-.376 1.566-.663 2.258h10.244c-.287-.692-.502-1.49-.663-2.258C12.134 8.197 12 6.628 12 6a4.002 4.002 0 00-3.203-3.92L8 1.917zM14.22 12c.223.447.481.801.78 1H1c.299-.199.557-.553.78-1C2.68 10.2 3 6.88 3 6c0-2.42 1.72-4.44 4.005-4.901a1 1 0 111.99 0A5.002 5.002 0 0113 6c0 .88.32 4.2 1.22 6z"
                fillRule="evenodd"
              />
            </svg>
          </li>
          <li>
            <a
              className=""
              href="/upgrade"
              onClick={[Function]}
              style={Object {}}
            >
              <button>
                Upgrade
              </button>
            </a>
          </li>
          <li>
            <div>
              <div>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12,8c-1.178,0-2,0.822-2,2s0.822,2,2,2s2-0.822,2-2S13.178,8,12,8z"
                    fill="none"
                  />
                  <path
                    d="M12,4c-4.337,0-8,3.663-8,8c0,2.176,0.923,4.182,2.39,5.641c0.757-1.8,2.538-3.068,4.61-3.068h2 c2.072,0,3.854,1.269,4.61,3.068C19.077,16.182,20,14.176,20,12C20,7.663,16.337,4,12,4z M12,14c-2.28,0-4-1.72-4-4s1.72-4,4-4 s4,1.72,4,4S14.28,14,12,14z"
                    fill="none"
                  />
                  <path
                    d="M13,16.572h-2c-1.432,0-2.629,1.01-2.926,2.354C9.242,19.604,10.584,20,12,20s2.758-0.396,3.926-1.073 C15.629,17.582,14.432,16.572,13,16.572z"
                    fill="none"
                  />
                  <path
                    d="M12,2C6.579,2,2,6.579,2,12c0,3.189,1.592,6.078,4,7.924V20h0.102C7.77,21.245,9.813,22,12,22s4.23-0.755,5.898-2H18 v-0.076c2.408-1.846,4-4.734,4-7.924C22,6.579,17.421,2,12,2z M8.074,18.927c0.297-1.345,1.494-2.354,2.926-2.354h2 c1.432,0,2.629,1.01,2.926,2.354C14.758,19.604,13.416,20,12,20S9.242,19.604,8.074,18.927z M17.61,17.641 c-0.757-1.8-2.538-3.068-4.61-3.068h-2c-2.072,0-3.854,1.269-4.61,3.068C4.923,16.182,4,14.176,4,12c0-4.337,3.663-8,8-8 s8,3.663,8,8C20,14.176,19.077,16.182,17.61,17.641z"
                  />
                  <path
                    d="M12,6c-2.28,0-4,1.72-4,4s1.72,4,4,4s4-1.72,4-4S14.28,6,12,6z M12,12c-1.178,0-2-0.822-2-2s0.822-2,2-2s2,0.822,2,2 S13.178,12,12,12z"
                  />
                </svg>
              </div>
              <ul>
                <li>
                  <img
                    alt="user avatar"
                    src=""
                  />
                  <div>
                    <p>
                      
                    </p>
                    <p>
                      @
                      
                    </p>
                  </div>
                </li>
                <div />
                <li>
                  <a
                    className=""
                    href="/create"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Write a story
                  </a>
                </li>
                <li>
                  Stories
                </li>
                <li>
                  Stats
                </li>
                <li>
                  Design your profile
                </li>
                <li>
                  Settings
                </li>
                <div />
                <li>
                  <a
                    className=""
                    href="/reading-list"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Reading List
                  </a>
                </li>
                <li>
                  Publications
                </li>
                <li>
                  Control your recommendations
                </li>
                <li>
                  Medium Partner Program
                </li>
                <div />
                <li>
                  Become a member
                </li>
                <li>
                  Help
                </li>
                <li>
                  Sign out
                </li>
                <div />
                <li>
                  Privacy Terms
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  <footer
    className="sc-hKwCoD hAyHIU"
  >
    <section
      className="sc-eCImvq ehgDKL"
    >
      <article
        className="sc-jRQAMF iNeymu"
      >
        <a
          aria-current="page"
          className="sc-gKckTs jOojji active"
          href="#"
          onClick={[Function]}
          style={Object {}}
        >
          Learn More.
        </a>
        <p>
          Medium is an open platform where 170 million readers come to find insightful and dynamic thinking. Here, expert and undiscovered voices alike dive into the heart of any topic and bring new ideas to the surface. 
          <a
            aria-current="page"
            className="sc-iCfLBT jXWveb active"
            href="#"
            onClick={[Function]}
            style={Object {}}
          >
            Learn more
          </a>
        </p>
      </article>
      <article
        className="sc-jRQAMF iNeymu"
      >
        <a
          aria-current="page"
          className="sc-gKckTs jOojji active"
          href="#"
          onClick={[Function]}
          style={Object {}}
        >
          Make Medium yours.
        </a>
        <p>
          Follow the writers, publications, and topics that matter to you, and you’ll see them on your homepage and in your inbox.
           
          <a
            aria-current="page"
            className="sc-iCfLBT jXWveb active"
            href="#"
            onClick={[Function]}
            style={Object {}}
          >
            Explore
          </a>
        </p>
      </article>
      <article
        className="sc-jRQAMF iNeymu"
      >
        <a
          aria-current="page"
          className="sc-gKckTs jOojji active"
          href="#"
          onClick={[Function]}
          style={Object {}}
        >
          Share your thinking.
        </a>
        <p>
          If you have a story to tell, knowledge to share, or a perspective to offer — welcome home. It’s easy and free to post your thinking on any topic. 
          <a
            aria-current="page"
            className="sc-iCfLBT jXWveb active"
            href="#"
            onClick={[Function]}
            style={Object {}}
          >
            Write on Medium
          </a>
        </p>
      </article>
    </section>
    <div
      className="sc-bdvvaa RAIwt"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <svg
          fill="white"
          height="29"
          width="145"
        >
          medium.svg
        </svg>
      </a>
      <ul
        className="sc-gsDJrp eBOCND"
      >
        <a
          aria-current="page"
          className="sc-dkPtyc iMpgWc active"
          href="#!"
          onClick={[Function]}
          style={Object {}}
        >
          About
        </a>
        <a
          aria-current="page"
          className="sc-dkPtyc iMpgWc active"
          href="#!"
          onClick={[Function]}
          style={Object {}}
        >
          Help
        </a>
        <a
          aria-current="page"
          className="sc-dkPtyc iMpgWc active"
          href="#!"
          onClick={[Function]}
          style={Object {}}
        >
          Legal
        </a>
      </ul>
    </div>
  </footer>,
]
`;
